lib/plugins/aws/package/compile/layers/index.test.js

```diff
@@ -118,6 +119,48 @@ describe('AwsCompileLayers', () => {
       });
     });
 
+    it('should create a layer resource with a retention policy', () => {
+      const s3Folder = awsCompileLayers.serverless.service.package.artifactDirectoryName;
+      const s3FileName = awsCompileLayers.serverless.service.layers.test.package.artifact
+        .split(path.sep).pop();
+      awsCompileLayers.serverless.service.layers = {
+        test: {
+          path: 'layer',
+          retain: true,
+        },
+      };
+      const compiledLayer = {
+        Type: 'AWS::Lambda::LayerVersion',
+        Properties: {
+          Content: {
+            S3Bucket: { Ref: 'ServerlessDeploymentBucket' },
+            S3Key: `${s3Folder}/${s3FileName}`,
+          },
+          LayerName: 'test',
+        },
+      };
+      const sha = crypto.createHash('sha1').update(JSON.stringify(compiledLayer)).digest('hex');
+      compiledLayer.DeletionPolicy = 'Retain';
```

Comment:
Nitpicking here but this could also be added to the `compiledLayer` object above to make it more "explicit".

Reply:
it's done here because in the real code the hash is computed from the object with out it added.

Reply:
Ahh, I see. That makes sense :+1: